Unable to rename exported variable in Go

In Go, exported variables (i.e., those accessible from other packages) must start with an uppercase letter. When using live rename, it seems the variable is temporarily renamed to a random string. However, if that temporary name doesn’t start with an uppercase letter, it throws the following error:

```
[LSP] rename, error getting references: `xxx.go:552:6: renaming "XXX" to "kajshfybcwriwuybqkjh" would make it unexported
xxx.go:524:14:	breaking references from packages such as "github.com/xxx/xxx"`
```

--- a/lua/live-rename.lua
+++ b/lua/live-rename.lua
@@ -440,6 +440,12 @@ function M.rename(opts)
         -- 2. doesn't conflict with other identiiers
         local unique_name = cword.text ~= "kajshfybcwriwuybqkjh" and "kajshfybcwriwuybqkjh" or "boviutyiiehsihjdlkgh"
 
+        -- match the casing of the word to be renamed as to not indroduce unwanted side effects
+        -- in some languages, like Go, uppercase identifiers are public
+        if cword.text:sub(1, 1):match("%u") then
+            unique_name = unique_name:sub(1, 1):upper() .. unique_name:sub(2)
+        end
+
         rename_params.newName = unique_name
         local handler = rename_refs_handler(transaction_id, unique_name)
         client:request(lsp_methods.textDocument_rename, rename_params, handler, doc_buf)

